### ISSUE TITLE
test module fails when using modules inside modules

### ISSUE THREAD

**Issue by rochana-atapattu** [suzuki-shunsuke/tfaction#1576]
### tfaction version

1.2.1

### Overview

when creating a new module that uses modules, we need to run `terraform init` before 
```yaml
- uses: suzuki-shunsuke/tfaction/test-module@72536e0b22961437dbca58cb4f3ea20a6b50b58f # v1.2.1
        with:
          github_token: ${{steps.generate_token.outputs.token}}
        env:
          AQUA_GITHUB_TOKEN: ${{github.token}}
```
step

### How to reproduce

tfaction-root.yaml

```yaml
---
plan_workflow_name: pull_request_target # Mandatory
label_prefixes:
  tfmigrate: "migrate:"
# skip_create_pr: true

# drift_detection:
#   # issue_repo_owner:
#   # issue_repo_name:
#   minimum_detection_interval: 5 # 5 days. By default 7 days.
#   num_of_issues: 3 # 3 working directories are checked per workflow run. By default 1 working directory is checked.

# tflint:
#   enabled: true
# tfsec:
#   enabled: false
# trivy:
#   enabled: true

# scaffold_working_directory:
#   skip_adding_aqua_packages: true

aqua:
  update_checksum:
    # Update aqua-checksums.json in `setup` action
    enabled: true # default is false
    skip_push: false # default is false
    prune: true # default is false

target_groups:
  - working_directory: aws/
    target: aws/
    aws_region: ap-northeast-1
    s3_bucket_name_tfmigrate_history: '<S3 Bucket Name for tfmigrate history files>'
    template_dir: templates/aws

    # terraform_plan_config:
    #   aws_assume_role_arn: arn:aws:iam::000000000000:role/GitHubActions_Terraform_AWS_terraform_plan
    # tfmigrate_plan_config:
    #   aws_assume_role_arn: arn:aws:iam::000000000000:role/GitHubActions_Terraform_AWS_tfmigrate_plan
    # terraform_apply_config:
    #   aws_assume_role_arn: arn:aws:iam::000000000000:role/GitHubActions_Terraform_AWS_terraform_apply
    # tfmigrate_apply_config:
    #   aws_assume_role_arn: arn:aws:iam::000000000000:role/GitHubActions_Terraform_AWS_tfmigrate_apply

  - working_directory: for-test/
    target: for-test/
    # aws_region: ap-northeast-1 # CHANGEME Fix the region if necessary
    # s3_bucket_name_tfmigrate_history: '<S3 Bucket Name for tfmigrate history files>' # CHANGEME
    template_dir: templates/for-test
    # terraform_plan_config:
    #   aws_assume_role_arn: arn:aws:iam::000000000000:role/GitHubActions_Terraform_github_terraform_plan # CHANGEME
    # tfmigrate_plan_config:
    #   aws_assume_role_arn: arn:aws:iam::000000000000:role/GitHubActions_Terraform_github_tfmigrate_plan # CHANGEME
    # terraform_apply_config:
    #   aws_assume_role_arn: arn:aws:iam::000000000000:role/GitHubActions_Terraform_github_terraform_apply # CHANGEME
    # tfmigrate_apply_config:
    #   aws_assume_role_arn: arn:aws:iam::000000000000:role/GitHubActions_Terraform_github_tfmigrate_apply # CHANGEME

  - working_directory: github/services/
    target: github/
    aws_region: ap-northeast-1 # CHANGEME Fix the region if necessary
    s3_bucket_name_tfmigrate_history: '<S3 Bucket Name for tfmigrate history files>' # CHANGEME
    template_dir: templates/github
    # terraform_plan_config:
    #   aws_assume_role_arn: arn:aws:iam::000000000000:role/GitHubActions_Terraform_github_terraform_plan # CHANGEME
    # tfmigrate_plan_config:
    #   aws_assume_role_arn: arn:aws:iam::000000000000:role/GitHubActions_Terraform_github_tfmigrate_plan # CHANGEME
    # terraform_apply_config:
    #   aws_assume_role_arn: arn:aws:iam::000000000000:role/GitHubActions_Terraform_github_terraform_apply # CHANGEME
    # tfmigrate_apply_config:
    #   aws_assume_role_arn: arn:aws:iam::000000000000:role/GitHubActions_Terraform_github_tfmigrate_apply # CHANGEME

  - working_directory: gcp/
    target: gcp/
    gcs_bucket_name_tfmigrate_history: '<GCS Bucket Name for tfmigrate history files>'
    template_dir: templates/gcp
    # terraform_plan_config:
    #   gcp_service_account: terraform-plan@example.iam.gserviceaccount.com
    #   gcp_workload_identity_provider: projects/000000000000/locations/global/workloadIdentityPools/github/providers/tfaction
    # tfmigrate_plan_config:
    #   gcp_service_account: terraform-plan@example.iam.gserviceaccount.com
    #   gcp_workload_identity_provider: projects/000000000000/locations/global/workloadIdentityPools/github/providers/tfaction
    # terraform_apply_config:
    #   gcp_service_account: terraform-apply@example.iam.gserviceaccount.com
    #   gcp_workload_identity_provider: projects/000000000000/locations/global/workloadIdentityPools/github/providers/tfaction
    # tfmigrate_apply_config:
    #   gcp_service_account: terraform-apply@example.iam.gserviceaccount.com
    #   gcp_workload_identity_provider: projects/000000000000/locations/global/workloadIdentityPools/github/providers/t
```

tfaction.yaml

```yaml

```

GitHub Actions Workflow

```yaml
---
name: Test Modules
on:
  workflow_call:
    inputs:
      modules:
        required: true
        type: string
      ref:
        required: true
        type: string
    secrets:
      gh_app_id:
        required: true
      gh_app_private_key:
        required: true
jobs:
  test-module:
    name: "test-module (${{matrix.target}})"
    runs-on: ubuntu-latest
    permissions:
      contents: read
    env:
      TFACTION_TARGET: ${{matrix.target}}
      TFACTION_JOB_TYPE: "terraform"
      GH_COMMENT_SHA1: ${{inputs.ref}}
      TFCMT_SHA: ${{inputs.ref}}
    strategy:
      fail-fast: true
      matrix:
        target: ${{fromJSON(inputs.modules)}}
    steps:
      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
        with:
          ref: ${{inputs.ref}}

      - name: Generate token
        id: generate_token
        uses: tibdex/github-app-token@3beb63f4bd073e61482598c45c71c1019b59b73a # v2.1.0
        with:
          app_id: ${{secrets.gh_app_id}}
          private_key: ${{secrets.gh_app_private_key}}
          # pull_requests:write - Post comments
          # contents:write - Push commits
          permissions: >-
            {
              "pull_requests": "write",
              "contents": "write"
            }
          repositories: >-
            ["${{github.event.repository.name}}"]

      - uses: aquaproj/aqua-installer@7c7338067bdb97d5bea2acc82b5870afca470d18 # v2.3.0
        with:
          aqua_version: v2.24.0
          aqua_opts: -l -a
          working_directory: ${{matrix.target}}
        env:
          AQUA_GITHUB_TOKEN: ${{github.token}}
      - uses: suzuki-shunsuke/tfaction/test-module@72536e0b22961437dbca58cb4f3ea20a6b50b58f # v1.2.1
        with:
          github_token: ${{steps.generate_token.outputs.token}}
        env:
          AQUA_GITHUB_TOKEN: ${{github.token}}

```

Other related code such as local Registry
main.tf
```
module "service_accounts" {
  source  = "terraform-google-modules/service-accounts/google"
  version = "4.2.2"

  for_each = local.service_accounts


  project_id    = var.project_id
  names         = [each.value.unique_name]
  project_roles = each.value.roles
  display_name  = "${each.value.name} for ${each.value.repo}"
  description   = "${each.value.description} for ${each.value.repo}"
}
```

GitHub Actions' log

```
 /home/runner/.local/share/aquaproj-aqua/bin/reviewdog -f rdjson -name tflint -filter-mode nofilter -reporter github-check -level warning -fail-on-error 1
reviewdog: Reporting results for "tflint"
Error: [tflint] reported by reviewdog 🐶


Raw Output:
location:{path:"main.tf" range:{start:{line:26} end:{line:26}}} severity:ERROR source:{name:"tflint" url:"https://github.com/terraform-linters/tflint"} code:{}
Error: "tflint failed"
```
this error is not properly logged in the job. i had to run the `tflint` cmd locally to find the issue
```
tflint --format json --modules
{"issues":[],"errors":[{"summary":"\"service_accounts\" module is not found. Did you run \"terraform init\"?","message":"","severity":"error","range":{"filename":"main.tf","start":{"line":26,"column":1},"end":{"line":26,"column":26}}}]}
```

### Expected behaviour

should test the module without error

### Actual behaviour

getting an error in `tflint` step

### Note

added
```yaml

      - name: terraform init
        run: terraform init --upgrade
        working-directory: ${{matrix.target}}
```
to fix the issue

**Comment by suzuki-shunsuke** [comment]
Thank you for your report!

https://github.com/terraform-linters/tflint/blob/66c7be71b6b18bbd4cd93ce843098acd5dc74642/terraform/loader.go#L126-L141

As you said, we have to run `terraform init` before running `tflint`.

**Comment by suzuki-shunsuke** [comment]
@rochana-atapattu

Could you try the prerelease version [v1.2.2-1](https://github.com/suzuki-shunsuke/tfaction/releases/tag/v1.2.2-1)?

- https://github.com/suzuki-shunsuke/tfaction/pull/1578

If there is no problem, I'll merge the pull request and release v1.2.2.

**Comment by rochana-atapattu** [comment]
tested with the patch and the pipeline was successful. thank you for the quick response.

**Comment by suzuki-shunsuke** [comment]
📝 

test-module ran `terraform init` at tfaction v0.7.1 or older, but at v1.0.0 we changed test-module not running `terraform init`.
I forgot the reason of the change, so I checked the change log.

- https://github.com/suzuki-shunsuke/tfaction/pull/1291
  - https://github.com/suzuki-shunsuke/tfaction/commit/d45bc33abd2cc06dac778c7a4544b2763810f0b3#diff-28f39e848ef8e864686c1149256dcf52c6cc7d3c7538d07c36148e0228f0790b
  - https://github.com/suzuki-shunsuke/tfaction/pull/1341

Unfortunately, I couldn't see the description about this change, but
I don't think we have the strong reason.
Probably we thought we don't have to run `terraform init` to test a Module.

I don't think there is any problem to run `terraform init` in test-module, so we can accept the change.
